### PR TITLE
check content_type via system call

### PR DESCRIPTION
**Pull request by lbiedinger**

### CODE REVIEW

File: app/uploaders/media_uploader.rb [hunk 1]
```diff
@@ -46,4 +48,17 @@ def store_dir
...
   #   "something.jpg" if original_filename
   # end
+
+  ##
+  # This overrides the default content_type which is based only on the file extension.
+  # Instead this sets the content_type by making a system call to inspect the mime-type.
+  #
+  def set_content_type
+    file_content_type = `file --b --mime-type '#{path}'`.strip
+    if file.respond_to?(:content_type=)
```
Comment (rwd) [inline app/uploaders/media_uploader.rb]: Why is this necessary? i.e. when will `file.respond_to?(:content_type=)` and when not?

Comment (lbiedinger) [reply app/uploaders/media_uploader.rb]: probably it's always available for any version of carrierwave we're using. I just took the code from here: http://www.rubydoc.info/github/carrierwaveuploader/carrierwave/CarrierWave%2FMimeTypes:set_content_type
(you'll have to 'view source' the 'github ' link doesn't work anymore)
Since the carrierwave gem natively checked this at some point, I thought better safe than sorry and  just took their check too. I don't know if 'file' can still somehow not implement 'content_type=' within the current carrierwave version.

Comment (rwd) [reply app/uploaders/media_uploader.rb]: Fair enough!
